dask/base.py

```diff
@@ -158,12 +158,9 @@ def annotate(**annotations):
         yield
 
 
-def is_dask_collection(x):
+def is_dask_collection(x) -> bool:
     """Returns ``True`` if ``x`` is a dask collection"""
-    try:
-        return x.__dask_graph__() is not None
-    except (AttributeError, TypeError):
-        return False
+    return hasattr(x, "__dask_graph__") and callable(x.__dask_graph__)
```

Comment:
@ian-r-rose there's a test in dask.array that `setattr`s `__dask_graph__` to `None`:

https://github.com/dask/dask/blob/7259f2150ab9c53409b848af3bf197a928a1de21/dask/array/tests/test_dispatch.py#L227-L236

I wonder if its in the scope of this PR to try to adopt a mechanism for "erasing" an existing `__dask_graph__` without setting it to `None`.

Reply:
To me, erasing an existing method on an instance is such specialized (and probably not recommended) behavior that we shouldn't try to capture it in the signature. I don't really even understand what this test is doing, to be honest.

Reply:
Yea I also can't quite tell what is being accomplished with the test, and see the assignment of the method to `None` as very specialized. But anyway, I'm satisfied with this definition of `is_dask_collection`, tests look to be passing and I think it covers what the previous implementation was getting (it while removing the possibility of a `None` return from the method itself).